dev/deps/spark-deps-hadoop-3-hive-2.3

```diff
@@ -33,6 +34,7 @@ breeze-macros_2.13/2.1.0//breeze-macros_2.13-2.1.0.jar
 breeze_2.13/2.1.0//breeze_2.13-2.1.0.jar
 bundle/2.24.6//bundle-2.24.6.jar
 cats-kernel_2.13/2.8.0//cats-kernel_2.13-2.8.0.jar
+checker-qual/3.42.0//checker-qual-3.42.0.jar
```

Comment:
Why was `checker-qual` added?

Reply:
This seems to be added here to add annotation.
- https://github.com/apache/arrow/issues/37286